package.json

```
@@ -14,7 +14,7 @@
   },
   "homepage": "https://github.com/microsoft/FluidExamples#readme",
   "engines": {
-    "node": ">=16.18.38"
+    "node": ">=18.0.0"
```

>18.0.0 [](http://example.com/codeflow?start=15&length=6)

```suggestion
18.7.1
```
since that is what we are working with and validate?

---

Where did you get that version? We now only validate node 20+, but at least the types dep should keep us from using features not in 18.0.0 directly.

---

Right - that is the version we had previously validated with.

---

It can be observed in pipeline logs.
[https://dev.azure.com/fluidframework/public/_build/results?buildId=190680&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=09732988-edb3-5e88-604f-3f1d71269f50](https://dev.azure.com/fluidframework/public/_build/results?buildId=190680&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=09732988-edb3-5e88-604f-3f1d71269f50)

---

If thats a requirement for the fluid server packages we consume, they can/should have their own engines block which will apply that restriction for any code that uses it which will be more robust, targeted and maintainable then making every use of it stay up to date via declaring its own limitations.

---

Agreed.
I was _suggesting_ the version here because that is what we had checked against. We expect but don't know that earlier versions of 18 are supported. We want people on 20+ so I don't think asking for recent 18 version is a problem.

---

BTW that log I referenced was FluidExamples - not FluidFramework proper.

---

I'd like to avoid rejecting users wanting to play with the examples because their node install is a bit old, so I just manually tested 18.0.0 locally and confirmed it works fine with at least the brainstorm sample. I also confirmed (before picking 18.0.0) that it had no patch releases, so it likely didn't have horrible problems.